tools/distpartitioning/globalids.py

```diff
@@ -0,0 +1,279 @@
+import numpy as np
+import torch
+import operator
+import itertools
+from gloo_wrapper import allgather_sizes, alltoall_cpu, alltoallv_cpu
+
+def get_shuffle_global_nids(rank, world_size, global_nids_ranks, node_data):
+    """ 
+    For nodes which are not owned by the current rank, whose global_nid <-> shuffle_global-nid 
+    is not present at the current rank, this function retrieves their shuffle_global_ids from the owning
+    rank
+
+    Parameters: 
+    -----------
+    rank : integer
+        rank of the process
+    world_size : integer
+        total no. of ranks configured
+    global_nids_ranks : list
+        list of numpy arrays (of global_nids), index of the list is the rank of the process
+                    where global_nid <-> shuffle_global_nid mapping is located. 
+    node_data : numpy ndarray, integers
+        node data with additional columns inserted
```

Comment:
you need to define the node data. it's not easy to understand. also, do you really need to provide all columns related to nodes? i suppose you only need to have shuffled global nids.

Reply:
As suggested, node_data is now a dictionary